lib/puppet/data_providers.rb

```diff
@@ -1,4 +1,6 @@
+# TODO: API 5.0, remove this module
```

Comment:
All the deprecations of the data providers (v4) are in the ticket PUP-6514. This commit should have its tag changed from PUP-6780 to PUP-6514.